src/main/common/filter.c

```diff
@@ -302,3 +302,34 @@ float firFilterDenoiseUpdate(firFilterDenoise_t *filter, float input)
         return filter->movingSum / ++filter->filledCount + 1;
     }
 }
+
+//Fast two-state Kalman
+void fastKalmanInit(fastKalman_t *filter, float q, float r, float p, float intialValue)
+{
+	filter->q     = q * 0.000001f; //add multiplier to make tuning easier
+	filter->r     = r * 0.001f;    //add multiplier to make tuning easier
+	filter->p     = p * 0.001f;    //add multiplier to make tuning easier
+	filter->x     = intialValue;   //set intial value, can be zero if unknown
+	filter->lastX = intialValue;   //set intial value, can be zero if unknown
+	filter->k     = 0.0f;          //kalman gain,  
+}
+
+float fastKalmanUpdate(fastKalman_t *filter, float input)
```

Comment:
This filter will quickly converge to constant value, it is only complicated way to implement simple RC (PT1) LP filter. Only difference is that tuning is performed using noise estimates instead of (easier to understand) cutoff frequency.